pymc3/step_methods/hmc/nuts.py

```diff
@@ -242,15 +239,15 @@ def _build_subtree(self, left, depth, epsilon):
 
         tree2, diverging, turning = self._build_subtree(tree1.right, depth - 1, epsilon)
 
-        size = tree1.size + tree2.size
+        size = np.logaddexp(tree1.size, tree2.size)
         accept_sum = tree1.accept_sum + tree2.accept_sum
         n_proposals = tree1.n_proposals + tree2.n_proposals
 
         left, right = tree1.left, tree2.right
         span = np.sign(epsilon) * (right.q - left.q)
         turning = turning or (span.dot(left.p) < 0) or (span.dot(right.p) < 0)
 
-        if bern(tree2.size * 1. / max(size, 1)):
+        if bern(np.exp(tree2.size - size)):
```

Comment:
nothing needed here/yet, but it looks like we might just refactor `bern` to work in log space, since that's how we're always using it now.